synapse/handlers/auth.py

```diff
@@ -2281,7 +2281,7 @@ async def on_logged_out(
         # call all of the on_logged_out callbacks
         for callback in self.on_logged_out_callbacks:
             try:
-                callback(user_id, device_id, access_token)
+                await callback(user_id, device_id, access_token)
```

Comment:
Argh are there any tests of the module API which could/should have caught this?

Related: https://github.com/python/mypy/issues/2499, https://bugs.python.org/issue30491

Reply:
Hmm good point, I'll add a test for this